subcommands/keys/ca_revoke_device_ca.go

```diff
@@ -0,0 +1,83 @@
+package keys
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/foundriesio/fioctl/x509"
+)
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "revoke-device-ca <PKI Directory>",
+		Short: "Revoke device CA from the list of CAs allowed to issue device client certificates",
+		Run:   doRevokeDeviceCa,
+		Args:  cobra.ExactArgs(1),
+		Long: `Revoke device CA from the list of CAs allowed to issue device client certificates.
+
+When the online or local device CA is revoked:
+- It is no longer possible to register new devices with client certificates issued by this device CA.
+- Existing devices with client certificates issued by this device CA can no longer connect to your FoundriesFactory.
+
+You may later re-add a revoked device CA using "keys ca update", if you still keep its certificate stored somewhere.
+Once you do it, devices with client certificates issued by this device CA may connect to your FoundriesFactory again.`,
```

Comment:
So, lets see if I understood this one:

- a) **First we create the CA (PKI)**: By running `fioctl keys ca create /path/to/cert/dirs -f <factory>`
- b) **Then, we can remove the CA(PKI)**: (which is introduced here) by running `fioctl key revoke-device-ca /path/to/cert/dirs -f <factory>` will revoke all that was scaffold in the dir with the first command. Would that mean:

     - All devices using this CA will lose update capabilities. 
     - Thus, revoking a CA in a production environment has the potential to affect all associated devices. 
     
      Can you please confirm my understanding here?

- c) **We can revert, restoring from Backup:** If a backup of the certificates is available, it seems we can restore them using `fioctl keys ca update <file>`. The documentation isn't entirely clear on whether we need to update all certificate files (factory_ca.key, factory_ca.pem, local-ca.key, local-ca.pem, online-crt, and tls-crt) individually, or if a subset will suffice. Any insights on this would be appreciated. See, from the fioctl keys ca update --help command:

```
Update the list of CAs that can create client certificates for devices

Usage:
  fioctl keys ca update <ca-crts file> [flags]
```

Reply:
* b) This command would only revoke a user-specified device CA (by file name or serial). Other device CAs are unaffected; factory root CA cannot be revoked. Then - yes, _devices using this CA will lose connectivity to FoundriesFactory_. However, bear in mind, that this command does not remove CA file on disk; it only removes the device CA on backend.
* c) Yes, a user can revert this operation _in case if they still possess the CA file they revoked_.
   For that the user would append that CA file to the list of allowed device CAs and upload it using `fioctl keys ca update`.
* e) No, the `fioctl keys ca create` can only be run once for a factory.